invertedai/logs/debug_logger.py

```diff
@@ -0,0 +1,77 @@
+from pydantic import BaseModel, validate_arguments
+from typing import List, Optional, Dict, Tuple
+from datetime import datetime
+
+import json
+
+class DebugLogger:
+
+    def __init__(self,debug_log_path):
+        self.debug_log_path = debug_log_path
+
+        self.data = {
+            "location_info_requests":[],
+            "location_info_responses":[],
+            "location_info_request_timestamps":[],
+            "location_info_response_timestamps":[],
+            "initialize_requests":[],
+            "initialize_responses":[],
+            "initialize_request_timestamps":[],
+            "initialize_response_timestamps":[],
+            "drive_requests":[],
+            "drive_responses":[],
+            "drive_request_timestamps":[],
+            "drive_response_timestamps":[],
+        }
+
+        file_name = "iai_log_" + self._get_current_time_human_readable_UTC() + "_UTC.json"
+        self.log_path = self.debug_log_path + file_name
+
+    def _create_directory(self):
+        if not os.path.isdir(self.debug_log_path):
+            os.makedirs(self.debug_log_path)
+            print(f"Debug log directory does not exist: Created new directory at {self.debug_log_path}")
+        print(f"Directory already exists!")
+
+    def append_request(self,model,data_dict):
```

Comment:
I am not so sure if "model" is the right name for parameter here. Maybe api_name ? And what do you think of having enum here instead of raw string for these api names.

Reply:
"model" is the name used in the "utils.py" Session class which is where this log is used so I named this parameter to align with that naming structure. I can change that in Session as well if you think that'll be worthwhile.

Reply:
I do like the idea of using an enum but I'm planning on making some changes to the log format that will impact both python and C++ SDK however this branch was meant to only replicate the C++ debug log into python.